### DISCUSSION TITLE
ram issue?

### DISCUSSION THREAD

**Discussion by cempro** [systemed/tilemaker#372] [Q&A] [answered]
Hello,
I am **trying to create mbtiles for the whole planet but only for the levels between 0 and 8.** 

As you know, whole planet OSM extract is about 62GB (pbf). **This may sound a bit weird but I do not need the road network objects in the mbtiles.**

So, **should I still need at least ~128GB ram to perform this operation?** If so, how can I create a subset from a PBF or OSM file which will not to include road layers/objects. I've seen Osmfilter tool but couldn't figure how to work with it.

Note: I will also include the externally downloaded shapefiles in mbtiles; coastline water polygon and landcover shapes ice, glaciated, urban areas (3files).

Thanks for your support!

**Comment by systemed** [comment, marked as answer]
`osmium tags-filter` is faster and easier to use than osmfilter: https://docs.osmcode.org/osmium/latest/osmium-tags-filter.html

You'll need to create a config file which lists the tags you want to retain, then call `osmium tags-filter -e configfile.txt -o outputfile.osm.pbf inputfile.osm.pbf`. This will create a new (smaller!) pbf which you can then use in tilemaker.

**Reply by cempro** [reply]
Thanks for quick response but I couldn't find a Osmium release for Windows, is there any Windows builds available?

Some suggests to use Osmosis -which seems to have 'reject' filter so that it will be easier just to omit roads and buildings- But, there is a discussion that Osmosis may not ne available that extracting more than 65k features (although this sounds weird!?) 

So, could you provide any URL for Osmium windows built if you know?

And does below command looks okay to run so that I could extract all types of features with the tags below?

`osmium tags-filter planet-latest.osm.pbf nwr/aerialway nwr/aeroway nwr/waterway nwr/water nwr/ocean nwr/aerodrome_label nwr/landuse nwr/landcover -o filtered_planet-latest.osm.pbf `

**Reply by systemed** [reply]
Sorry, I don't have any Windows expertise. You might be better off using osmfilter/osmconvert in that case.

Whichever you use, you need to filter on OSM tags (because they're what's contained in the .pbf file). Your suggested command is trying to use the name of vector tile layers which won't work.

**Reply by cempro** [reply]
Forgive me if I go too far. Is it possible for you to provide me a sample command line to run with Osmium-tool (I am going to run it under live-ubuntu) so that I can exclude buildings and roads to reduce/repack planet-pbf size.

Does Value column stand for OSM TAGs?
https://wiki.openstreetmap.org/wiki/Map_features#Aerialway

**Reply by systemed** [reply]
OSM tags are key/value pairs, e.g. highway=primary, name=High Street, ref=A34. You need to find out what tags are used for the information you want to keep. [This page](https://wiki.openstreetmap.org/wiki/Map_features) details the most common ones. You probably want to keep things like rivers (i.e. the `waterway` tag), lakes (`natural`), landuse (`landuse`) and so on.

**Reply by cempro** [reply]
> Only .pbf.

Can I exclude buildings and highways with the below command with Osmium? 

```
osmium tags-filter -i city.osm.pbf wr/building  w/ highway ​-o no-buildings-roads.osm.pbf
``

Or could you please provide an example multiple filter command (either to include or to exclude.)

**Comment by cempro** [comment]
Just another simple question, does Tilemaker also support *.osm as input or it only accepts *pbf ?

**Reply by systemed** [reply]
Only .pbf.

**Reply by cempro** [reply]
Thank you! Can you suggest a tool for converting OSM to PBF. I started a filter task using Osmfilter and it will create a filtered osm output.

**Reply by systemed** [reply]
osmconvert will do that, I think.

**Reply by SomeoneElseOSM** [reply]
An example of osmconvert is https://github.com/SomeoneElseOSM/Boundary_Scripts/blob/master/update_boundaries.sh#L139 (though that's a Linux shell script, and you're only interested in the osmfilter / osmconvert lines).

**Comment by dieterdreist** [comment]
I suggest you create an extract of a small area and perform some testing.
Generally my expectation would be that you will want to exclude much more
tags than you want to keep, so it might make more sense to go with a
positive list.
